src/tests/CMakeLists.txt

```diff
@@ -23,17 +24,19 @@ target_link_libraries(${tgt} ${libraries_common})
 add_gtest_to_target(${tgt})
 
 
-set(tgt "streamer_test")
-set(sources
-streamer_test.cxx
-$<TARGET_OBJECTS:kafka_to_nexus__objects>
-)
-add_executable(${tgt} ${sources})
-add_dependencies(${tgt} flatbuffers_generate)
-target_compile_definitions(${tgt} PRIVATE ${compile_defs_common})
-target_include_directories(${tgt} PRIVATE ${path_include_common})
-target_link_libraries(${tgt} ${libraries_common})
-add_gtest_to_target(${tgt})
+#set(tgt "streamer_test")
```

Comment:
What is going on here, should these test changes be in this branch? If I uncomment this I see `streamer_test.cxx` seems to be broken.

Reply:
The main reason is that just a very small part of the Streamer can be tested in a unit test, since it requires Kafka to do almost anything. I'm still looking for a workaround to run without Kafka (that's why a few days ago I asked if you have some Kafka mock).
Since this PR introduces relevant changes  for the integration with NIKOS I'd like to defer the refactoring of the test in another PR.

Reply:
Ok. For Mantid I used mocks and fakes of the Kafka interface. Here are the fakes:
https://github.com/mantidproject/mantid/blob/master/Framework/LiveData/test/KafkaTesting.h
and some of the tests are here:
https://github.com/mantidproject/mantid/blob/master/Framework/LiveData/test/KafkaEventStreamDecoderTest.h
If you go with this approach it might make sense to do it after Mark's suggested refactoring of KafkaW into separate consumer and producer.